hudi-common/src/main/java/org/apache/hudi/common/util/ObjectSizeCalculator.java

```diff
@@ -384,9 +383,69 @@ public int getArrayHeaderSize() {
 
           @Override
           public int getObjectHeaderSize() {
+            return 4;
+          }
+
+          @Override
+          public int getObjectPadding() {
+            return 4;
+          }
+
+          @Override
+          public int getReferenceSize() {
+            return 4;
+          }
+
+          @Override
+          public int getSuperclassFieldPadding() {
+            return 4;
+          }
+        };
+      } else {
+        // it's a 64-bit uncompressed references object model
+        return new MemoryLayoutSpecification() {
+          @Override
+          public int getArrayHeaderSize() {
+            return 16;
+          }
+
+          @Override
+          public int getObjectHeaderSize() {
+            return 16;
+          }
+
+          @Override
+          public int getObjectPadding() {
+            return 8;
+          }
+
+          @Override
+          public int getReferenceSize() {
+            return 8;
+          }
+
+          @Override
+          public int getSuperclassFieldPadding() {
+            return 8;
+          }
+        };
+      }
+    } else {
```

Comment:
Guess this was the original code. that only handled hotspot vm

Reply:
right, the code for hotspot vm remained the same